[Finishes #187355127] prompt user to change password afert x amout of time

Celestin4 opened the pull request:
### 1. What does this PR do?

User should be prompted to change their passwords every x amount of time 
since the time it was either manually updated.

**### 2. Description of task to be completed**

Why:
The app needs to be secure enough. when passwords update is not enforced,
it becomes a loophole to security

Description
As a user <Any>
I should be prompted to update my password
Such that the information about the last time password reset be recorded

Acceptance criteria :
Given user created/password updated, the notification should be sent with respect with the time

Scenario 1
when the notification prompt is worked on, the timer restarts for the next
time of update

Scenario 2
when the notification prompt is ignored on, the update password prompt
should be turned
on and no further operation is allowed.

<><><><>

Then:

field in the user /profile model should contain last_time_password updated
Dev notes:

Use Crone jobs to iterate though all users checking if the password
expiration has reached
Use Node events to implement the password update event.
Environmemt variable should contain time for password expiration
Relevant pivotal tracker stories
https://www.pivotaltracker.com/story/show/187355091

**### 3. How should this be manually tested?**

- Clone the repository using git clone 

- Checkout to the branch using `git checkout <branch-name>`

- Install node packages that the application depends on: `npm i`

- Create a .env file in the root directory and add all environment variables required to run the application. The required variables are listed in the .env.example file available in root directory of the application.

- Start the application development server: `npm run dev`


### Screenshoot

Tests

<img width="1015" alt="image" src="https://github.com/atlp-rwanda/e-commerce-ones-and-zeroes-bn/assets/97307891/e3ec16d2-b484-4b94-a74d-ce94731c68c0">


Sending emails for updating password

<img width="1065" alt="image" src="https://github.com/atlp-rwanda/e-commerce-ones-and-zeroes-bn/assets/97307891/903a9922-8bdc-4317-8b71-d6083266ff81">

UmuhireJessie commented on src/helps/cron.job.ts:
```diff
@@ -0,0 +1,27 @@
+        const users = await db.User.findAll();
+        
+        users.forEach(async (user: any) => {
+            const lastPasswordChangeDate: Date = user.passwordLastChanged;
```
I am not seeing where this field is being changed. How is it helping you to know when is the password last updated?

Celestin4 replied:
It has been added to the user model. It will also be added to the token on login. It will also be updated on updating the password.

UmuhireJessie commented on src/middleware/changePasswordIgnored.ts:
```diff
@@ -0,0 +1,32 @@
+            const currentTime = new Date();
+            const oneDayInMillis = 24 * 60 * 60 * 1000;
+
+            if ((currentTime.getTime() - lastPasswordChangeDate.getTime()) > oneDayInMillis &&
```
Why are you doing this? What if I have set that the password should be changed after 7 days

Celestin4 replied:
This was previous commit for testing. Now we are reading it from .env

UmuhireJessie reviewed (changes requested):
@Celestin4 check my comments, also add some tests

UmuhireJessie reviewed (changes requested):
@Celestin4 please add:
1. Tests
2. screenshots of successful operation

Celestin4 committed:
[Finishes #187355127] prompt user to change password afert x amout of time

rebasing 3

tesing tests

testing file

fixing missing files

fixing configuration

removing unwanted console

removing unwanted console

UmuhireJessie commented on src/middleware/changePasswordIgnored.ts:
```diff
@@ -0,0 +1,41 @@
+          oneDayInMillis &&
+        req.path !== '/login'
+      ) {
+        console.log('You are required to update your password');
+        console.log(
+          'You are now redirected to https://onesandzeroes/users/update',
+        );
+
```
Remove all debugging consoles.

UmuhireJessie reviewed (changes requested):
Check my comments

Christian-Ishimwe reviewed (changes requested):
Instead of putting cron.Job.ts  in helps, can you create it's own folder for only CronJobs? in addition, since it related to prompt the user for changing password, change it's name from cron.job.ts to like password.job.ts, since there is other cron Jobs
